src/main/java/org/hydev/mcpm/client/database/mirrors/MirrorSelector.java

```diff
@@ -47,4 +50,18 @@ public void updateMirrors() throws IOException
         LOCAL_PATH.getParentFile().mkdirs();
         Files.writeString(LOCAL_PATH.toPath(), yml);
     }
+
+    @Override
+    public Mirror getSelectedMirror() throws IOException
+    {
+        if (LOCAL_SEL.exists()) return YML.readValue(Files.readString(LOCAL_SEL.toPath()), new TypeReference<>() {});
+        return listAvailableMirrors().get(0);
+    }
+
+    @Override
+    public void setSelectedMirror(Mirror mirror) throws IOException
+    {
+        LOCAL_SEL.getParentFile().mkdirs();
+        Files.writeString(LOCAL_SEL.toPath(), YML.writeValueAsString(mirror));
```

Comment:
What level is MirrorSelector.java in CA? Is it database and frameworks? Makes sense.

Reply:
Yea I think it should be